web-lib-funcs.pl

```diff
@@ -12997,6 +12997,126 @@ sub get_default_system_locale
 return $locale_def;
 }
 
+=head2 get_http_redirect(url, [page], [timeout])
+
+Check if given URL redirects somewhere
+
+The parameters are :
+
+=item url - Given URL to check if it redirects anywhere, e.g. https://google.com
+
+=item page - Path in URL, e.g. /about as in https://www.google.com/about
+
+=item timeout - Timeout for connections, defaults to 30s
+
+=cut
+
+sub get_http_redirect
+{
+my ($url, $page, $timeout) = @_;
+my ($out, $error, $con_err);
+my ($proto, $host, $uport) = $url =~ /^(https?):\/\/([^:\/?#]*)(?:\:([0-9]+))?/;
+my ($ssl, $port) = (0, undef);
+$port = '80' if (!$uport);
+if ($proto eq 'https') {
+	$ssl = 1;
+	$port = '443' if (!$uport);
+	}
+
+# Set default port, page and timeout
+$port ||= $uport;
+$page ||= '/';
+$timeout ||= 30;
+
+# Original request
+my %rs;
+$rs{'url'} = $url;
+$rs{'proto'} = $proto;
+$rs{'host'} = $host;
+$rs{'port'} = $port;
+$rs{'page'} = $page;
+eval "use LWP::UserAgent";
+if (!$@) {
```

Comment:
Why not just always use webmin's built-in http download code if you're going to have to support that case anyway?

Reply:
> Well, at first Webmin code in this PR needs to be able to handle multiple redirects. Also, I just thought that using `LWP::UserAgent` is more straight forward, as it is very well supported and many users use it..

Although, I think it doesn't work the same with the local connections. Perhaps, it's indeed better to rely only on Webmin download code?

Reply:
Best to use only the Webmin code to keep things simpler